massa-final-state/src/final_state.rs

```diff
@@ -666,23 +710,24 @@ impl FinalState {
     pub fn is_db_valid(&self) -> bool {
         let db = self.db.read();
 
-        for (serialized_key, serialized_value) in db.iterator_cf(STATE_CF, MassaIteratorMode::Start)
+        // check if the execution trial hash is present and valid
         {
-            if !serialized_key.starts_with(CYCLE_HISTORY_PREFIX.as_bytes())
-                && !serialized_key.starts_with(DEFERRED_CREDITS_PREFIX.as_bytes())
-                && !serialized_key.starts_with(ASYNC_POOL_PREFIX.as_bytes())
-                && !serialized_key.starts_with(EXECUTED_OPS_PREFIX.as_bytes())
-                && !serialized_key.starts_with(EXECUTED_DENUNCIATIONS_PREFIX.as_bytes())
-                && !serialized_key.starts_with(LEDGER_PREFIX.as_bytes())
-                && !serialized_key.starts_with(MIP_STORE_PREFIX.as_bytes())
-            {
-                warn!(
-                    "Key/value does not correspond to any prefix: serialized_key: {:?}, serialized_value: {:?}",
-                    serialized_key, serialized_value
-                );
+            let execution_trail_hash_serialized: [u8] =
+                match db.get(EXECUTION_TRAIL_HASH_PREFIX.as_bytes()) {
```

Comment:
Should be ok with get_cf() and a call to .expect() to unwrap potential DB read errors.

Reply:
corrected, without expect to really have a true is_valid function :)